packages/shared/routes/dashboard/wallet/views/Send.svelte

```diff
@@ -98,27 +120,150 @@
+        switch(type) {
+            default:
+                if(ledgerAwaitingConfirmation) {
+                    ledgerAwaitingConfirmation = false
+
+                    closePopup()
+                }
+
+                break
+
+            case TransferProgressEventType.GeneratingRemainderDepositAddress:
+                transactionEventData = data
+
+                /**
+                 * NOTE: The break statement is omitted in this case to allow the next block of code
+                 * (under SigningTransaction) to be executed.
+                 */
+
+            case TransferProgressEventType.SigningTransaction:
+                ledgerAwaitingConfirmation = true
+
+                openPopup({
+                    type: 'ledgerTransaction',
+                    hideClose: true,
+                    props: {
+                        onCancel: _onCancel,
+                        ...handleTransactionEventData(transactionEventData)
+                    }
+                })
+
+                break
+
+            case TransferProgressEventType.PreparedTransaction:
+                /**
+                 * CAUTION: The Ledger confirmation doesn't always trigger
+                 * the popup to close, so it is programmatically enforced here.
+                 */
+                if(get(popupState).active)
+                    closePopup()
+
+                transactionEventData = data
+
+                break
         }
     }
+
+    $: if(!$isSoftwareProfile)
+        handleTransferState($transferState)
+
     $: if (!$isTransferring && ledgerAwaitingConfirmation) {
         closePopup()
     }
 
+    const checkLedgerDeviceState = (state: LedgerDeviceState, notificationType: NotificationType = 'error', ignoreNotDetected: boolean = false): void => {
+        /**
+         * NOTE: The NotDetected state is another state to not show notifications for
+         * because the Ledger app assumes this state upon entering / exiting apps for a
+         * miniscule amount of time (but still enough to read as "Not Detected"), resulting
+         * in lots of unnecessary errors. The logic here is a little hard to reason, but it
+         * accomodates for if we want to ignore the NotDetected state.
+         */
+        switch(state) {
+            case LedgerDeviceState.Connected:
+                break
+
+            case LedgerDeviceState.NotDetected:
+                if(ignoreNotDetected)
+                    break
+
+            case LedgerDeviceState.Locked:
+                if(transactionTimeoutId)
+                    clearTimeout(transactionTimeoutId)
+
+                transactionTimeoutId = setTimeout(() => checkLedgerDeviceState(get(ledgerDeviceState), notificationType, ignoreNotDetected), 10000)
+
+            default:
+                const message = locale(`error.ledger.${state}`)
+
+                const numSimilarNotifications = get(displayNotifications).filter((nd: NotificationData) => nd.type === notificationType).length
```

Comment:
`const notificationIsNew = get(displayNotifications).filter((nd: NotificationData) => nd.type === notificationType).length === 0` would be easier to read